src/main/java/wolox/training/models/User.java

```diff
@@ -0,0 +1,85 @@
+package wolox.training.models;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToMany;
+import javax.persistence.SequenceGenerator;
+import wolox.training.exceptions.BookAlreadyOwnedException;
+
+public class User {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "USER_SEQ")
+    @SequenceGenerator(name = "USER_SEQ", sequenceName = "USER_SEQ")
+    private long id;
+
+    @Column(nullable = false)
+    private String username;
+
+    @Column(nullable = false)
+    private String name;
+
+    @Column(nullable = false)
+    private LocalDate birthDate;
+
+    @ManyToMany(cascade = CascadeType.ALL)
```

Comment:
This is not the correct CascadeType. By saying ALL, you are saying: Persist, Merge, Refresh, Delete and Detach.

Detach is only so that something gets detached along the cascade when you detach an object from the persistence context manually. This almost never happens, so we won't focus on this point.

Persist says: If you persist the Users instance to the DB, then persist every item that this collection has in the DB as well.

Delete says: If you delete the Users instance from the DB, then delete every book item that this collection has in the DB as well.

Merge says: If you changed the in-memory values of the collection, then merge this changes to the DB so these are reflected there.

Refresh says: If something changed in the database values of the collection, then refresh those changes in the in-memory values.

[Here](https://www.baeldung.com/jpa-cascade-types) is a link explaining it.

In this case, we just need MERGE and REFRESH.
You can look at MERGE as the instruction to "merge" changes on the model back to the database.
But we're lacking the one that brings modifications iof the database to the model. And that's REFRESH :)